pkg/envoy/sds/response.go

```diff
@@ -145,58 +144,66 @@ func (s *sdsImpl) getRootCert(cert certificate.Certificater, sdscert envoy.SDSCe
 		return secret, nil
 	}
 
-	// Program SAN matching based on SMI TrafficTarget policies
-	switch sdscert.CertType {
-	case envoy.RootCertTypeForMTLSOutbound:
+	svcIdentities, err := getServiceIdentitiesFromCert(sdsCert, s.serviceIdentity, s.meshCatalog)
+	if err != nil {
+		return nil, err
+	}
+
+	secret.GetValidationContext().MatchSubjectAltNames = getSubjectAltNamesFromSvcIdentities(svcIdentities)
+	return secret, nil
+}
+
+func getServiceIdentitiesFromCert(sdsCert envoy.SDSCert, svcIdentity identity.ServiceIdentity, mc catalog.MeshCataloger) ([]identity.ServiceIdentity, error) {
```

Comment:
Do you think an extra unit test for this function makes sense? I know it implicitly gets tested when other functions that call it are tested.

Reply:
Not yet. This is an intermediary step to further simplification.  I would not test this just yet.